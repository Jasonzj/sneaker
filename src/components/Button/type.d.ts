export type Props = {
  text: string
  type: 'button' | 'submit' | 'reset' | undefined
  loading: boolean
}
