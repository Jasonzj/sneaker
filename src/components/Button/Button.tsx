import React from 'react'
import style from './button.module.css'
import { Props } from './type'

const Button: React.FC<Props> = ({ text, type, loading }) => {
  return (
    <button type={type} className={style.button} disabled={loading}>
      <div className={style.content}>{loading ? 'Loading...' : text}</div>
    </button>
  )
}

export default React.memo(Button)
