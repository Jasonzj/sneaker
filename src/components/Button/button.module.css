.button {
  position: relative;
  display: block;
  margin-top: 20px;
  width: 100%;
  font-size: 16px;
  font-weight: bold;
  color: #fff;
  line-height: 30px;
  text-align: center;
  background-color: #49a9ee;
  border: 1px solid transparent;
}

.button:disabled :before {
  pointer-events: none;
  content: '';
  position: absolute;
  left: -1px;
  top: -1px;
  right: -1px;
  bottom: -1px;
  border-radius: inherit;
  background-color: hsla(0, 0%, 100%, 0.35);
}

.button:hover {
  background: #2fa8ff;
}

.content {
  display: block;
  height: 30px;
  width: 100%;
}
