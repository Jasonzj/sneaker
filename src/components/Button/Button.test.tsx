import { render, screen } from '@testing-library/react'
import Button from './Button'
import { Props } from './type'

const props: Props = {
  text: 'Login In',
  type: 'submit',
  loading: false,
}

test('should render correctly text', () => {
  const { asFragment } = render(<Button {...props} />)

  expect(screen.getByText('Login In')).toBeInTheDocument()
  expect(asFragment()).toMatchSnapshot()
})

test('should render correctly type', () => {
  render(<Button {...props} />)

  expect(screen.getByRole('button')).toHaveProperty('type', 'submit')
})

test('should render correctly loading status', () => {
  const { rerender, asFragment } = render(<Button {...props} />)

  expect(screen.queryByText(/Loading/i)).not.toBeInTheDocument()

  rerender(<Button {...props} loading={true} />)

  expect(screen.getByText(/Loading/i)).toBeInTheDocument()
  expect(asFragment()).toMatchSnapshot()
})
