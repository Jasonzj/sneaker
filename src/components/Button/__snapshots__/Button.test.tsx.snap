// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
    type="submit"
  >
    <div
      class="content"
    >
      Login In
    </div>
  </button>
</DocumentFragment>
`;
